HPXMLtoOpenStudio/resources/defaults.rb

```diff
@@ -6111,102 +6124,133 @@ def self.get_water_heater_heating_capacity(fuel, nbeds, num_water_heaters, nbath
+    # FUTURE: Incorporate number of occupants for conventional elec/gas storage WHs.
+
     if nbaths.nil?
       nbaths = Defaults.get_num_bathrooms(nbeds)
     end
 
-    if fuel != HPXML::FuelTypeElectricity # Non-electric tank WHs
-      case nbeds
-      when 0, 1, 2
-        return 30.0
-      when 3
-        if nbaths <= 1.5
-          return 30.0
-        else
-          return 40.0
-        end
-      when 4
-        if nbaths <= 2.5
-          return 40.0
-        else
-          return 50.0
-        end
-      else
+    if is_hpwh && !n_occ.nil? # Heat pump water heater
+      # Source: Jeff Maguire recommendation for ResStock when num occupants is known
+      if (nbeds <= 2) && (n_occ <= 3)
+        # Up to 2 bedrooms AND up to 3 occupants
         return 50.0
+      elsif (nbeds >= 4) || (n_occ >= 5)
+        # 4 or more bedrooms OR 5 or more occupants
+        return 80.0
+      else
+        # (3 bedrooms AND up to 4 occupants) OR (up to 3 bedrooms AND 4 occupants)
+        return 66.0
       end
-    else
-      case nbeds
-      when 0, 1
-        return 30.0
-      when 2
-        if nbaths <= 1.5
-          return 30.0
-        else
-          return 40.0
-        end
-      when 3
-        if nbaths <= 1.5
-          return 40.0
-        else
-          return 50.0
-        end
-      when 4
-        if nbaths <= 2.5
-          return 50.0
+
+    elsif fuel == HPXML::FuelTypeElectricity # Electric tank WHs
+      # Source: Table 8. Benchmark DHW Storage and Burner Capacity in 2014 BA HSP
+      if nbeds <= 1
+        tank_vol = 30.0
+      elsif nbeds <= 2
+        tank_vol = (nbaths <= 1.5 ? 30.0 : 40.0)
+      elsif nbeds <= 3
+        tank_vol = (nbaths <= 1.5 ? 40.0 : 50.0)
+      elsif nbeds <= 4
+        tank_vol = (nbaths <= 2.5 ? 50.0 : 66.0)
+      elsif nbeds <= 5
+        tank_vol = 66.0
+      else
+        tank_vol = 80.0
+      end
+
+      if is_hpwh
+        # Bump up size (e.g., 30/40 => 50, 50 => 66, 66 => 80)
+        # Note: There are very few HPWHs < 50 or > 80 gal in AHRI
+        if tank_vol < 50
+          tank_vol = 50.0
+        elsif tank_vol < 66
+          tank_vol = 66.0
         else
-          return 66.0
+          tank_vol = 80.0
         end
-      when 5
-        return 66.0
+      end
+
+      return tank_vol
+
+    else # Non-electric tank WHs
+      # Source: Table 8. Benchmark DHW Storage and Burner Capacity in 2014 BA HSP
+      if nbeds <= 2
+        return 30.0
+      elsif nbeds <= 3
+        return (nbaths <= 1.5 ? 30.0 : 40.0)
+      elsif nbeds <= 4
+        return (nbaths <= 2.5 ? 40.0 : 50.0)
       else
-        return 80.0
+        return 50.0
       end
     end
   end
 
+  # Gets the assumed COP of the water heater's heat pump based on UEF regressions.
+  #
+  # @param water_heating_system [HPXML::WaterHeatingSystem] The HPXML water heating system of interest
+  # @return [Double] COP of the heat pump (W/W)
+  def self.get_water_heater_heat_pump_cop(water_heating_system)
```

Comment:
Just doing a little code refactor (code moved from `waterheater.rb`), this is unrelated and doesn't change anything.